build.gradle

```diff
@@ -329,38 +329,51 @@ configurations {
 }
 dependencies {
     // https://mvnrepository.com/artifact/org.plumelib/require-javadoc
-    compileOnly group: 'org.plumelib', name: 'require-javadoc', version: '0.1.2'
+    requireJavadoc group: 'org.plumelib', name: 'require-javadoc', version: '0.1.2'
 }
-task requireJavadoc(type: Javadoc, group: "Documentation") {
-    description = 'Ensures that Java elements have Javadoc documentation.'
-    destinationDir = file("${rootDir}/docs/tmpapi")
-    destinationDir.mkdirs()
-    source(project(':checker').sourceSets.main.allJava, project(':framework').sourceSets.main.allJava,
-            project(':dataflow').sourceSets.main.allJava, project(':javacutil').sourceSets.main.allJava)
-    classpath = files(subprojects.collect { it.sourceSets.main.compileClasspath })
 
-    destinationDir.deleteDir()
 
-    // options.memberLevel = JavadocMemberLevel.PRIVATE
-    options.docletpath = project.sourceSets.main.compileClasspath as List
-    options.doclet = "org.plumelib.javadoc.RequireJavadoc"
-    // options.addStringOption('skip', 'ClassNotToCheck|OtherClass')
-}
-task requireJavadocPrivate(type: Javadoc, group: "Documentation") {
-    description = 'Ensures that all (even private) Java elements have Javadoc documentation.'
-    destinationDir = file("${rootDir}/docs/tmpapi")
-    destinationDir.mkdirs()
-    source(project(':checker').sourceSets.main.allJava, project(':framework').sourceSets.main.allJava,
-            project(':dataflow').sourceSets.main.allJava, project(':javacutil').sourceSets.main.allJava)
-    classpath = files(subprojects.collect { it.sourceSets.main.compileClasspath })
+/**
+ * Creates a task named taskName that runs javadoc.
+ *
+ * @param taskName the name of the task to create
+ * @param memberLevel the JavadocMemberLevel to use
+ * @param requireJavadoc whether or not to run the RequireJavadoc doclet. If false -Xdoclint:all is
+ * passed. Defaults to true.
+ * @return the new task
+ */
+def createJavadocTask(taskName, memberLevel, requireJavadoc = true) {
+    tasks.create(name: taskName, type: Javadoc) {
+        description = 'Ensures that Java elements have Javadoc documentation.'
```

Comment:
I would take the description as a command-line argument; otherwise all the tasks have the identical documentation and it's hard for a user to understand the differences among them.